tests/conftest.py

```diff
@@ -0,0 +1,19 @@
+def pytest_runtest_setup(item):
+    """
+    Per-test setup.
+    - Provide cache isolation incrementing the cache key prefix
+    """
+    increment_cache_key_prefix()
+
+
+def increment_cache_key_prefix():
```

Comment:
Yeah, I cribbed that code from treeherder. But you're right, just clearing the cache is the obvious solution. I have no idea why we used this complicated piece of code there --  looks like it was actually added by you? :)

https://github.com/mozilla/treeherder/commit/9fd2f645429b2883a1097f9ddb933a34de49e359

/cc @edmorley